src/ack_manager.rs

```diff
@@ -41,10 +41,12 @@ pub struct AckManager {
     received: MessageFilter<Ack>,
 }
 
+/// An identifier for a waiting-to-be-acknowledged message (a hash of the message).
 #[derive(Clone, Copy, Eq, Hash, Ord, PartialEq, PartialOrd, RustcDecodable, RustcEncodable)]
 pub struct Ack(u64);
 
 impl AckManager {
+    /// Create a new manager, with empty lists.
```

Comment:
Let's use the third person singular present indicative ("Creates") in method docs:
https://aturon.github.io/style/comments.html#sentence-structure
(I do like the idea of using questions for `-> bool` methods, though, if nobody objects.)